src/Runner.Common/JobServer.cs

```diff
@@ -32,20 +35,21 @@ public sealed class JobServer : RunnerService, IJobServer
         private VssConnection _connection;
         private TaskHttpClient _taskClient;
 
-        public async Task ConnectAsync(VssConnection jobConnection)
+        public async Task ConnectAsync(Uri jobServerUrl, VssCredentials jobServerCredential, IThrottlingReporter jobServerQueue = null)
         {
-            _connection = jobConnection;
+            Trace.Info($"Establishing connection for JobServer");
             int attemptCount = 5;
-            while (!_connection.HasAuthenticated && attemptCount-- > 0)
+
+            while (attemptCount-- > 0)
             {
                 try
                 {
-                    await _connection.ConnectAsync();
+                    RefreshConnection(jobServerUrl, jobServerCredential, jobServerQueue);
                     break;
```

Comment:
i think we still need to call `await _connection.ConnectAsync();`

Reply:
or you can make `RefreshConnection` to call `_connection.ConnectAsync()`, we need the client to hit the server.
only `_connection.ConnectAsync()` and `_connection.GetClient<TaskHttpClient>();` can do that.